packages/webpack-cli/src/webpack-cli.ts

```diff
@@ -2478,6 +2488,10 @@ class WebpackCLI implements IWebpackCLI {
           this.logger.raw(printedStats);
         }
       }
+
+      if (options.failOnWarnings && (stats.hasErrors() || stats.hasWarnings())) {
```

Comment:
We should merge this with the condition above - https://github.com/webpack/webpack-cli/pull/3317/files#diff-fbc9bd0ae2f54d7f45607dc7979f8c1721de2e9ccf4a5934504bab29e7204805R2429

Reply:
Yeah, I was doing that only at first but this - https://github.com/webpack/webpack-cli/pull/3317#discussion_r909393891